### ISSUE TITLE
 More games versions that surpass WHDload.xml

### ISSUE THREAD

**Issue by bizzar721** [midwan/amiberry#405]
I manually cross referenced about 300 `.lha` files with the version listed in `whdload_db.xml`.   Below are the file names and SH1 of the games that have a newer version available.  I hope you find this information useful.

```
Alcatraz_v1.3_2073.lha
7c0197761e2af6c5a9ec06ca31b31935127a5812

AlienBreed_v2.3a_0998.lha
9aa6724131263425368c9fb7599fb70f33873001

Ambermoon_v2.0
467498408dbc76b4356e83a37cb1111f3d5ff6eb

Banshee_v3.3_CD32.lha
7ed5eb571843a084dc8935461ae29b947ccfead4

BehindTheIronGate_v1.1.lha
7041815f615d8f113702a43dcaaefcddaf6b1938

Carcharodon_v1.1a_2389.lha
884cb9b6b3639c4b141850eecc968dc11ffebbaf

Coala_v2.0_AGA_0242.lha
3dcc5cb2b57d0001036564f6be187645cda3809e

CosmicPirate_v1.2.lha
f0ba98a05f12fbf989494e792b93aa3b23e224c8

Damocles_v1.3_NTSC_1192.lha
7bf33c998b1582834d1ca2c56c498c61208fe280

DefenderOfTheCrown_v3.1_NTSC_0902.lha
1930eac5470916570ef32381f8eeccf74a37c78a

DynaBlaster_v1.3_0664.lha
7e57bfe22dfcc28146ba2b98a668ed7f1eac8982

Ebonstar_v1.0_NTSC_0915.lha
57528508e55d8662851ae8058dd7cb48ab780ae6

Falcon_v1.1_NTSC_1197.lha
102b2d023c6741b2ca7f2436f441037991601cc4

GreatGianaSisters_v1.6_2945.lha
294570ec82486b615e49782e0b4dafa0435a159d

Hacker_v1.0_NTSC_0906.lha
0d04dc8193b28d5913c8c199e23d19ae35edebac

Hybris_v2.3_NTSC_1457.lha
9e73047d80ecb3fd3fc5e5125bc8d3ec42b70537

ImpossibleMission2025_v2.0-B_1107.lha
9c7df0348d71a66e35a78a771010a81d81624fff

MindWalker_v0.9_NTSC_0951.lha
3ff66beabaa63467d95d08589185be1652c6d02d

Odyssey_v2.1_2854.lha
6794265a17f34668307c06ab9fce889e33ad3b99

RuffNTumble_v2.3_0199.lha
31488a65e0ad395d1f9cf91a35a81a6048c3e63f

Speedball2_v2.3_CD32.lha
26e6c5f58a514d55f20657b1a789fcd6ccff48e5

ThunderCats_v1.2_1783.lha
f57903e4af2362dbf7b99698789f9e3a07e2bf05

ThunderHawkAH73M_v2.2.1_0039.lha
9f3af85e6b0bea7ab4ff8db832927815d2c098af

WormsDirectorsCut_v1.3_AGA_0605.lha
5440fa5310dd2f99e68711532b673995569602c4

ZombieApocalypse2_v1.1_AGA.lha
2ca59a6b7f8325a4dc934b1f1e68938277d30c5d
```

I am having trouble fully understanding the WHDload database.  I have been having trouble finding documentation that explains it.  I was expecting to see the latest game version as well as all previous versions listed in the WHDload database.  Since only the last known version is in the database, would that mean updating Amiberry can potentially break compatibility  for any games that got updated?

I am also interested in what is needed / how to add the new game versions into the database. Any links you can point my way, I would really appreciate.

**Comment by bizzar721** [comment]
I will check it out and give any reports if needed.  

So WHD load database `xml` lists  the game ID, some basic settings, variables/overrides? Everything else such as controller mapping & other enhancements are self-contained within the `.lha`?

Technically, could I copy & paste just the entries for the game versions above and all work as intended?

Thank you for your time.  I really appreciate your help.

**Comment by HoraceAndTheSpider** [comment]
Some settings can be found from when the LHA/Slave file is scanned.... but not all are sadly.

Controller mapping is set by the user, but it is possible to include some 'good practice' suggestions for this in the XML, if I am told what people want for them. (see Legend/EOB etc which have these)

I dont suggest copy/paste, but helping to get the XML auto-updated correctly, as then the change will feed into weekly updates.

**Comment by bizzar721** [comment]
I am more than happy to help. I mentioned copy/paste more so for a point of reference for my own learning.  I didn't know if there were any other game specific files besides the `.lha` & database entry.  

Will all game versions be included in the WHDload database (so it doesn't break compatability for older versions)?

Last thing, what steps would you recommend I do to make sure I am thorough in helping to get the XML auto-updated correctly.

**Comment by HoraceAndTheSpider** [comment]
Sorry, probably I  explained myself badly.

You are right,  and  each entry has settings which are defined by the “scanning” of that file. 

In many cases the scanner doesn’t need to know about a version being different before it is able to apply the right settings, but it does need the new file to read. It uses a look up against the “sub path” inside the LHA (eg “MonkeyIsland2” to see if the scanner knows about that particular game. If it does, it makes sure those settings are retained in the XML for any version scanned. 

 So basically I have a Pi dedicated to regularly syncing the LHA files from a certain FTP source, (and it retains old versions) and re-running the scanner every few days and re-uploading once a week. 

I’ve asked @midwan to test the latest XML in Amiberry because I’ve made a few changes in the format. If that works you’ll be able to download the latest xml file from my repo and just replace your existing one.

**Comment by bizzar721** [comment]
Awesome. Thank you and the rest of the team for the hard work and taking the time to explain.

**Comment by midwan** [comment]
The latest version of the DB is now added in `master`

**Comment by bizzar721** [comment]
A couple of games had issues.  I'm not at my pc right now, but off the top of my head, DynaBlaster had sound but no picture. That can be fixed by holding down both mouse buttons right after runcommand loading screen goes away to access Amiga menu and change display mode to NTSC.

The other errors had the same issue as banshee where it wasn't loading AGA chipset (caola & impossible mission 2025 were 2 of them).

A fix was to create a custom config with the options listed in `WHDloader.xml`, so it is not loading the chipset option using WHDloader. 

I downloaded Amiberry from source roughly 2 weeks ago in a fresh build and havent changed any configurations except replaced `WHDload.xml`.

What would be the best course of action? Is it something I could fix and submit, or should I submit it as a bug report? (or could it just possably something borked with my install?)  

Thanks, have a good weekend.

**Comment by midwan** [comment]
@bizzar721 I think the best approach would be to submit any improvements to the DB on @HoraceAndTheSpider 's repo, so it can be updated from there.

Perhaps @HoraceAndTheSpider can comment further on this?

**Comment by HoraceAndTheSpider** [comment]
All of the above games are added to the XML. 

On the next 'sweep' the XML will be setting `NTSC=TRUE` for all games which have `NTSC` in the file name... eg. `Falcon_v1.1_NTSC_1197.lha` etc

**Comment by HoraceAndTheSpider** [comment]
Leaving this open becuase of the NTSC factor. It appears to be in the XML but possible not set by Amiberry.

**Comment by HoraceAndTheSpider** [comment]
NTSC detection should be fixed now with the latest XML.

**Comment by HoraceAndTheSpider** [comment]
NTSC detection issue fixed.
